### PR TITLE
feat: optionally disable config bootstrap

### PR DESCRIPTION
**Pull request by diivi**
[#725](https://projects.torsion.org/borgmatic-collective/borgmatic/issues/725)

### CODE REVIEW

**Comment by diivi** [comment]
Nevermind, it still stored the config 🤔

**Review by witten — CHANGES_REQUESTED** [review]
Looks great, but would you also mind adding some text coverage for this? It could be a very basic unit test that exercises the non-default code path. Thank you!

**Comment by witten** [comment]
> Nevermind, it still stored the config

Huh, weird. This totally looks like it'd work. Maybe you're not using the latest archive when looking for the stored config?

**Comment by witten** [comment]
Oh, maybe this changeset only prevents storage of the manifest but doesn't change whether the actual config files get stored?

**Comment by diivi** [comment]
> Oh, maybe this changeset only prevents storage of the manifest but doesn't change whether the actual config files get stored?

Yup

> Looks great, but would you also mind adding some test coverage for this? It could be a very basic unit test that exercises the non-default code path. Thank you!

On it!

File: tests/unit/borg/test_create.py [hunk 1]
```diff
@@ -646,6 +646,53 @@ def test_create_archive_with_sources_and_used_config_paths_calls_borg_with_sourc
         global_arguments=flexmock(log_json=False, used_config_paths=['/etc/borgmatic/config.yaml']),
     )
 
+def test_create_archive_with_sources_and_used_config_paths_with_store_config_files_false_calls_borg_with_sources_and_no_config_paths():
```
Comment (diivi) [inline tests/unit/borg/test_create.py:649]: Disclaimer: this is Copilot generated. I just took a glance (lines 691 and 677) and it made sense + it passed.

**Comment by witten** [comment]
The test looks fine to me (even though that function name makes even _me_ blush), but what about a test for the manifest code path?

**Comment by diivi** [comment]
Done!

**Comment by witten** [comment]
Awesome, thank you!

**Pull request merged**